openpype/hosts/maya/plugins/publish/extract_arnold_scene_source.py

```diff
@@ -113,34 +125,58 @@ def process(self, instance):
         instance.data["representations"].append(representation)
 
     def _extract(self, nodes, attribute_data, kwargs):
-        self.log.info("Writing: " + kwargs["filename"])
+        self.log.info(
+            "Writing {} with:\n{}".format(kwargs["filename"], kwargs)
+        )
         filenames = []
+        nodes_by_id = defaultdict(list)
         # Duplicating nodes so they are direct children of the world. This
         # makes the hierarchy of any exported ass file the same.
-        with delete_after() as delete_bin:
+        with lib.delete_after() as delete_bin:
             duplicate_nodes = []
             for node in nodes:
+                # Only interested in transforms:
+                if cmds.nodeType(node) != "transform":
+                    continue
+
+                # Only interested in transforms with shapes.
+                shapes = cmds.listRelatives(
+                    node, shapes=True, noIntermediate=True
+                ) or []
```

Comment:
No need for `or []` because we're continuing on `not shapes` anyway. Not sure if we care, but if we do.. then this could be the change.
```suggestion
                shapes = cmds.listRelatives(
                    node, shapes=True, noIntermediate=True
                )
```